ed448/src/field/scalar.rs

```diff
@@ -0,0 +1,1106 @@
+        let (_, underflow) = candidate.0.sbb(&ORDER, Limb::ZERO);
+        let underflow = Choice::from((underflow.0 >> (Limb::BITS - 1)) as u8);
+        CtOption::new(candidate, underflow & is_valid)
+    }
+
+    /// Serialize the scalar into 57 bytes, per RFC 8032.
+    /// Byte 56 will always be zero.
+    pub fn to_bytes_rfc_8032(&self) -> ScalarBytes {
+        let mut bytes = ScalarBytes::default();
+        bytes[..56].copy_from_slice(&self.to_bytes());
+        bytes
+    }
+
+    /// Construct a `Scalar` by reducing a 912-bit little-endian integer
+    /// modulo the group order ℓ.
+    pub fn from_bytes_mod_order_wide(input: &WideScalarBytes) -> Scalar {
+        // top multiplier = 2^896 mod ℓ
+        const TOP_MULTIPLIER: U448 = U448::from_be_hex("3402a939f823b7292052bcb7e4d070af1a9cc14ba3c47c44ae17cf725ee4d8380d66de2388ea18597af32c4bc1b195d9e3539257049b9b60");
+        let value = (
+            U448::from_le_slice(&input[..56]),
+            U448::from_le_slice(&input[56..112]),
+        );
+        let mut top = [0u8; 56];
+        top[..2].copy_from_slice(&input[112..]);
+        let top = U448::from_le_slice(&top).mul_mod(&TOP_MULTIPLIER, &NZ_ORDER);
+        let bottom = U448::rem_wide_vartime(value, &NZ_ORDER);
+        Self(bottom.add_mod(&top, &ORDER))
+    }
+
+    /// Construct a Scalar by reducing a 448-bit little-endian integer modulo the group order ℓ
+    pub fn from_bytes_mod_order(input: &ScalarBytes) -> Scalar {
+        let value = U448::from_le_slice(&input[..56]);
+        Self(value.rem_vartime(&NZ_ORDER))
+    }
+
+    /// Return a `Scalar` chosen uniformly at random using a user-provided RNG.
+    ///
+    /// # Inputs
+    ///
+    /// * `rng`: any RNG which implements the `RngCore + CryptoRng` interface.
+    ///
+    /// # Returns
+    ///
+    /// A random scalar within ℤ/lℤ.
+    pub fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
+        let mut scalar_bytes = WideScalarBytes::default();
+        rng.fill_bytes(&mut scalar_bytes);
+        Scalar::from_bytes_mod_order_wide(&scalar_bytes)
+    }
+
+    /// Computes the hash to field routine according to Section 5
+    /// <https://datatracker.ietf.org/doc/rfc9380/>
+    /// and returns a scalar.
+    ///
+    /// # Errors
+    /// See implementors of [`ExpandMsg`] for errors:
+    /// - [`ExpandMsgXmd`]
+    /// - [`ExpandMsgXof`]
+    ///
+    /// `len_in_bytes = <Self::Scalar as FromOkm>::Length`
+    ///
+    /// [`ExpandMsgXmd`]: elliptic_curve::hash2curve::ExpandMsgXmd
+    /// [`ExpandMsgXof`]: elliptic_curve::hash2curve::ExpandMsgXof
+    pub fn hash<X>(msg: &[u8], dst: &[u8]) -> Self
+    where
+        X: for<'a> ExpandMsg<'a>,
+    {
+        let mut random_bytes = Array::<u8, U84>::default();
+        let dst = [dst];
+        let mut expander =
+            X::expand_message(&[msg], &dst, random_bytes.len()).expect("invalid dst");
+        expander.fill_bytes(&mut random_bytes);
+        Self::from_okm(&random_bytes)
+    }
```

Comment:
The [specification says](https://www.rfc-editor.org/rfc/rfc9496.html#name-scalar-field-2) to use 64 bytes here. An external method taking 64 bytes, like `Scalar::from_bytes_mod_order_wide()`, would probably be useful as well.

Reply:
My assessment was incorrect here, the specification part I linked here is not related to the `hash_to_field` implementation.